Assets/Mirror/Runtime/MessagePacker.cs

```diff
@@ -84,9 +84,16 @@ public static T Unpack<T>(byte[] data) where T : IMessageBase, new()
         // -> NetworkReader will point at content afterwards!
         public static bool UnpackMessage(NetworkReader messageReader, out int msgType)
         {
-            // read message type (varint)
-            msgType = (int)messageReader.ReadUInt16();
-            return true;
+            try
+            {
+                msgType = (int)messageReader.ReadUInt16();
+                return true;
+            }
+            catch (System.IO.EndOfStreamException)
+            {
+                msgType = -1;
```

Comment:
msgTypes are casted to ushorts in few places, beware with negatives

Reply:
@rodolphito why do we catch this one instead of letting NetworkConnection.TransportReceive catch+disconnect?

Reply:
Then change it to void

Reply:
be careful about changing this to void,  this method is public,  so you could break games that might be calling this directly

Reply:
Methods that always return true are really awful code style